task_sdk/tests/execution_time/test_supervisor.py

```diff
@@ -328,6 +328,10 @@ def handle_request(request: httpx.Request) -> httpx.Response:
 
         client = make_client(transport=httpx.MockTransport(handle_request))
 
+        # Patch the kill method so we can assert it was called with the correct signal
+        # without waiting for escalation_delay seconds
+        mock_kill = mocker.patch("airflow.sdk.execution_time.supervisor.WatchedSubprocess.kill")
```

Comment:
SIGKILL shouldn't ever need to escalate, it'll hard kill the process and should finish up within milliseconds... 🤔

Reply:
I need to change a few things -- will remove this into separate PR.

For reference, it is because we wait for that many seconds before we close it here.

https://github.com/apache/airflow/blob/edf3e3376811129a9fad278ae396b3b5644c69d8/task_sdk/src/airflow/sdk/execution_time/supervisor.py#L436